util/socketset/socketset.go

```diff
@@ -0,0 +1,219 @@
+// Mgmt
+// Copyright (C) 2013-2018+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// Package socketset is in API for creating a select style netlink
+// socket to recieve events from the kernel.
+package socketset
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"syscall"
+
+	errwrap "github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
+// SocketSet is used to receive events from a socket and shut it down cleanly
+// when asked. It contains a socket for events and a pipe socket to unblock
+// receive on shutdown.
+type SocketSet struct {
+	fdEvents int
+	fdPipe   int
+	pipeFile string
+}
+
+// NewSocketSet returns a socketSet, initialized with the given parameters.
+func NewSocketSet(groups uint32, name string, proto int) (*SocketSet, error) {
+	fdEvents, err := unix.Socket(unix.AF_NETLINK, unix.SOCK_RAW, proto)
+	if err != nil {
+		return nil, errwrap.Wrapf(err, "error creating netlink socket")
+	}
+
+	if err := unix.Bind(fdEvents, &unix.SockaddrNetlink{
+		Family: unix.AF_NETLINK,
+		Groups: groups,
+		Pid:    uint32(os.Getpid()),
```

Comment:
This line and other changes (eg: changed comments, etc) are fine, but should be in a separate commit. So it might be best to have commit (1): move the socketSet without making any changes except the pub/priv stuff (2): edit the socketSet in-place with the changes you wanted to make to it. (3) use the new socket set in commit three with our fact.

Additionally this makes it easier to review because the diff of changes will be separate from the move.